src/backend/impl/src/repositories/types/proposal_review_commit.rs

```diff
@@ -0,0 +1,176 @@
+    storable::{Blob, Bound},
+    Storable,
+};
+
+use super::{CommitSha, DateTime, ProposalReviewId, UserId, Uuid};
+
+pub type ProposalReviewCommitId = Uuid;
+
+#[derive(Debug, CandidType, Deserialize, Clone, PartialEq, Eq)]
+pub enum ReviewCommitState {
+    Reviewed {
+        matches_description: bool,
+        comment: Option<String>,
+        highlights: Vec<String>,
+    },
+    NotReviewed,
+}
+
+#[derive(Debug, CandidType, Deserialize, Clone, PartialEq, Eq)]
+pub struct ProposalReviewCommit {
+    pub proposal_review_id: Uuid,
+    pub user_id: Uuid,
+    pub created_at: DateTime,
+    pub commit_sha: CommitSha,
+    pub state: ReviewCommitState,
+}
+
+impl ProposalReviewCommit {
+    pub fn is_reviewed(&self) -> bool {
+        matches!(&self.state, ReviewCommitState::Reviewed { .. })
+    }
+
+    pub fn is_not_reviewed(&self) -> bool {
+        matches!(&self.state, ReviewCommitState::NotReviewed)
+    }
+}
+
+impl Storable for ProposalReviewCommit {
+    fn to_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(Encode!(self).unwrap())
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        Decode!(bytes.as_ref(), Self).unwrap()
+    }
+
+    const BOUND: Bound = Bound::Unbounded;
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ProposalReviewCommitProposalReviewUserKey(Blob<{ Self::MAX_SIZE as usize }>);
+
+impl ProposalReviewCommitProposalReviewUserKey {
+    const MAX_SIZE: u32 = <(
+        (ProposalReviewId, (UserId, CommitSha)),
+        ProposalReviewCommitId,
+    )>::BOUND
+        .max_size();
+
+    pub fn new(
+        proposal_review_id: ProposalReviewId,
+        user_id: UserId,
+        commit_sha: CommitSha,
+        proposal_review_commit_id: ProposalReviewCommitId,
+    ) -> Result<Self, ApiError> {
+        Ok(Self(
+            Blob::try_from(((proposal_review_id, (user_id, commit_sha)), proposal_review_commit_id).to_bytes().as_ref()).map_err(
+                |_| {
+                    ApiError::internal(&format!(
+                        "Failed to convert proposal review id {:?}, user id {:?} and proposal review commit id {:?} to bytes.",
+                        proposal_review_id, user_id, proposal_review_commit_id
+                    ))
+                },
+            )?,
+        ))
+    }
+}
+
+pub struct ProposalReviewCommitProposalReviewUserRange {
+    start_bound: ProposalReviewCommitProposalReviewUserKey,
+    end_bound: ProposalReviewCommitProposalReviewUserKey,
+}
+
+impl ProposalReviewCommitProposalReviewUserRange {
+    pub fn new(
+        proposal_review_id: ProposalReviewId,
+        user_id: Option<UserId>,
+        commit_sha: Option<CommitSha>,
+    ) -> Result<Self, ApiError> {
+        Ok(Self {
+            start_bound: ProposalReviewCommitProposalReviewUserKey::new(
+                proposal_review_id,
+                user_id.unwrap_or(Uuid::min()),
+                commit_sha.unwrap_or(CommitSha::min()),
+                Uuid::min(),
```

Comment:
```suggestion
                ProposalReviewCommitId::min(),
```
Is a bit easier to read IMO